src/test/java/org/opendatakit/common/persistence/QueryResultTest.java

```diff
@@ -361,10 +363,21 @@ public void testCase3() throws ODKDatastoreException {
         int fetchSizes[] = {1, 2, 1021, 303, 101, 2831 };

```

Comment:
Agreed that this test is kinda weird. Since `fetchSizes` and `idxFetch` are used elsewhere, I would have expected a more narrow fix to be where you add 10 to the `fetchSizes[]` and update `idxFetch` to 5. Then maybe add a comment that about this being governed by SET_SIZE up above.